models/miamicoin-token-v2.model.ts

```diff
@@ -0,0 +1,173 @@
+import { Account, ReadOnlyFn, Tx, types } from "../deps.ts";
+import { Model } from "../src/model.ts";
+
+enum ErrCode {
+  ERR_UNAUTHORIZED = 2000,
+  ERR_TOKEN_NOT_ACTIVATED,
+  ERR_TOKEN_ALREADY_ACTIVATED,
+  ERR_V1_BALANCE_NOT_FOUND,
+  ERR_CORE_CONTRACT_NOT_FOUND = 6009,
+}
+
+export class MiamiCoinTokenModelV2 extends Model {
+  name = "miamicoin-token-v2";
+  static readonly ErrCode = ErrCode;
+
+  //////////////////////////////////////////////////
+  // SIP-010 FUNCTIONS
+  //////////////////////////////////////////////////
+
+  transfer(
+    amount: number,
+    from: Account,
+    to: Account,
+    sender: Account,
+    memo: ArrayBuffer | undefined = undefined
+  ): Tx {
+    let memoVal: string;
+
+    if (typeof memo == "undefined") {
+      memoVal = types.none();
+    } else {
+      memoVal = types.some(types.buff(memo));
+    }
+
+    return this.callPublic(
+      "transfer",
+      [
+        types.uint(amount),
+        types.principal(from.address),
+        types.principal(to.address),
+        memoVal,
+      ],
+      sender.address
+    );
+  }
+
+  getName(): ReadOnlyFn {
+    return this.callReadOnly("get-name");
+  }
+
+  getSymbol(): ReadOnlyFn {
+    return this.callReadOnly("get-symbol");
+  }
+
+  getDecimals(): ReadOnlyFn {
+    return this.callReadOnly("get-decimals");
+  }
+
+  getBalance(user: Account): ReadOnlyFn {
+    return this.callReadOnly("get-balance", [types.principal(user.address)]);
+  }
+
+  getTotalSupply(): ReadOnlyFn {
+    return this.callReadOnly("get-total-supply");
+  }
+
+  getTokenUri(): ReadOnlyFn {
+    return this.callReadOnly("get-token-uri");
+  }
+
+  //////////////////////////////////////////////////
+  // TOKEN CONFIGURATION
+  //////////////////////////////////////////////////
+
+  activateToken(sender: Account, stacksHeight: number): Tx {
+    return this.callPublic(
+      "activate-token",
+      [types.principal(sender.address), types.uint(stacksHeight)],
+      sender.address
+    );
+  }
+
+  //////////////////////////////////////////////////
+  // UTILITIES
+  //////////////////////////////////////////////////
+
+  setTokenUri(sender: Account, newUri?: string | undefined): Tx {
+    let newUriVal: string;
+
+    if (typeof newUri == "undefined") {
+      newUriVal = types.none();
+    } else {
+      newUriVal = types.some(types.utf8(newUri));
+    }
+
+    return this.callPublic(
+      "set-token-uri",
+      [newUriVal],
+      sender.address
+    );
+  }
+
+  mint(amount: number, recipient: Account, sender: Account): Tx {
+    return this.callPublic(
+      "mint",
+      [types.uint(amount), types.principal(recipient.address)],
+      sender.address
+    );
+  }
+
+  // burn(amount: number, owner: Account, sender: Account): Tx {
+  //   return this.callPublic(
+  //     "burn",
+  //     [types.uint(amount), types.principal(owner.address)],
+  //     sender.address
+  //   );
+  // }
```

Comment:
I believe this should be un-commented and used in at least 1 test.

Reply:
good catch, fixed in https://github.com/citycoins/contracts/pull/190/commits/cf7a02b04ddda068df13907bee3e86cf57144520

Since `miamicoin-token-v2` has the same burn function as nyc, I used the same set of tests from `newyorkcitycoin-token-v2`.